### DISCUSSION TITLE
Getting Fetch Error from CoPilot

### DISCUSSION THREAD

**Discussion by jnegron357** [community/community#116432] [Copilot Conversations] [answered]
### Select Topic Area

Bug

### Body

I'm using Copilot with the latest version of visual studio and I get this error whenever it tries to suggest any code: **statusNotification: Warning: unable to get local issuer certificate**. I'm currently in the trial phase and so far I'm not crazy about the product. Here is info on my VS install:

Microsoft Visual Studio Professional 2022
Version 17.9.5
VisualStudio.17.Release/17.9.5+34723.18
Microsoft .NET Framework
Version 4.8.04084

Installed Version: Professional

ASP.NET and Web Tools   17.9.199.22661
ASP.NET and Web Tools

Azure App Service Tools v3.0.0   17.9.199.22661
Azure App Service Tools v3.0.0

Azure Functions and Web Jobs Tools   17.9.199.22661
Azure Functions and Web Jobs Tools

Bundler & Minifier   2.9.9
Adds support for bundling and minifying JavaScript, CSS and HTML files in any project.

C# Tools   4.9.0-3.24121.1+a98c90d56455379836dd5c845b35fa932b00cfa3
C# components used in the IDE. Depending on your project type and settings, a different version of the compiler may be used.

Code alignment   15.1.118.0
Code alignment helps you present your code beautifully, enhancing clarity and readability.

Common Azure Tools   1.10
Provides common services for use by Azure Mobile Services and Microsoft Azure Tools.

GitHub Copilot   1.156.0.0 (v1.156.0.0@82fafb995)
GitHub Copilot is an AI pair programmer that helps you write code faster and with less work.

GitHub Copilot Agent   1.156.0

Markdown Editor v2   2.0.153
A full featured Markdown editor with live preview and syntax highlighting. Supports GitHub flavored Markdown.

Microsoft JVM Debugger   1.0
Provides support for connecting the Visual Studio debugger to JDWP compatible Java Virtual Machines

NuGet Package Manager   6.9.2
NuGet Package Manager in Visual Studio. For more information about NuGet, visit https://docs.nuget.org/

Razor (ASP.NET Core)   17.9.2.2412004+684740676513b3b208d5bd5b7f8786dab8d287b7
Provides languages services for ASP.NET Core Razor.

Snippet Designer   1.8.1
Snippet Designer is a Visual Studio plug in which allows you to create and search for snippets inside the IDE

SQL Server Data Tools   17.9.55.1
Microsoft SQL Server Data Tools

TypeScript Tools   17.0.30103.2001
TypeScript Tools for Microsoft Visual Studio

Visual Basic Tools   4.9.0-3.24121.1+a98c90d56455379836dd5c845b35fa932b00cfa3
Visual Basic components used in the IDE. Depending on your project type and settings, a different version of the compiler may be used.

Visual F# Tools   17.9.0-beta.23614.3+a521e1cd420beb56c15faf6836184fadd2b7937a
Microsoft Visual F# Tools

Visual Studio IntelliCode   2.2
AI-assisted development for Visual Studio.

Web Compiler   1.14.11
Compiler for LESS, Sass and CoffeeScript files

Here is the full error message from Visual Studio's output window:
GitHub Copilot 1.156.0.0 (v1.156.0.0@82fafb995)
GitHub Copilot Agent 1.156.0 (1.156.0)
[DEBUG] [agent] [2024-04-01T13:49:38.935Z] Agent service starting
[DEBUG] [agent] [2024-04-01T13:49:39.242Z] Telemetry initialized
Auth Status: OK
Successfully authorized GitHub Copilot.
[DEBUG] [getCompletions] [2024-04-01T13:53:04.680Z] Requesting completion at position 11:0, between "    {\r\n" and "\r\n    }\r\n".
[INFO] [default] [2024-04-01T13:53:05.321Z] [fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex
statusNotification: Warning: unable to get local issuer certificate
[ERROR] [ghostText] [2024-04-01T13:53:06.251Z] Error on ghost text request: (FetchError) unable to get local issuer certificate
[INFO] [RequestProposalsAsync] 0 completions available.

**Comment by davecheney** [comment, marked as answer]
Is it possible that your employer, internet service provider, or government, may be intercepting your SSL traffic?

**Reply by jnegron357** [reply]
My employer uses Zscaler for monitoring network traffic. Do they need to white list a domain to make this work?

**Reply by davecheney** [reply]
Yes, please see https://docs.github.com/en/copilot/copilot-business/enabling-and-setting-up-github-copilot-business#configuring-network-settings
